recipes/leveldb/all/conanfile.py

```diff
@@ -0,0 +1,71 @@
+import os
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+
+class LevelDBCppConan(ConanFile):
+    name = "leveldb"
+    description = "LevelDB is a fast key-value storage library written at Google that provides an ordered mapping from string keys to string values."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/leveldb"
+    topics = ("conan", "leveldb", "db")
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False], 
+        "with_snappy": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_snappy": True,
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.with_snappy:
+            self.requires("snappy/1.1.8")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["LEVELDB_BUILD_TESTS"] = False
+        self._cmake.definitions["LEVELDB_BUILD_BENCHMARKS"] = False
+        self._cmake.configure()
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
+        if not self.options.shared:
+            if self.settings.os == "Linux":
+                self.cpp_info.system_libs = ["pthread"]
```

Comment:
```suggestion
        if self.settings.os == "Linux":
            self.cpp_info.system_libs = ["pthread"]
```

Keep it as usual

Reply:
@madebr told me to do so?

Reply:
Above

Reply:
I checked and there is no mention of pthread in the headers.
So this should be just fine.

Reply:
ldd shows pthread, @gocarlos , keep it as usual

Reply:
You need to run `objdump -p`.
See the `security` section of `ldd`:
```
   Security
       Be  aware  that  in  some circumstances (e.g., where the program specifies an ELF interpreter other than ld-
       linux.so), some versions of ldd may attempt to obtain the dependency information by attempting  to  directly
       execute the program, which may lead to the execution of whatever code is defined in the program's ELF inter‐
       preter, and perhaps to execution of the program itself.  (In glibc versions before 2.27,  the  upstream  ldd
       implementation did this for example, although most distributions provided a modified version that did not.)

       Thus,  you  should  never  employ  ldd on an untrusted executable, since this may result in the execution of
       arbitrary code.  A safer alternative when dealing with untrusted executables is:

           $ objdump -p /path/to/program | grep NEEDED

       Note, however, that this alternative shows only the direct dependencies of the executable, while  ldd  shows
       the entire dependency tree of the executable.

```

Reply:
good tip! that's why I use containers or vms haha

Reply:
The last line is important: `objdump -p` shows only the direct dependencies of the executable, while  ldd  shows  the entire dependency tree of the executable